### PR TITLE
[ASCII-2608] Add workflow to update golang.org/x deps

### PR DESCRIPTION
**Pull request by pgimalac**
### What does this PR do?
Add a workflow to regularly update `golang.org/x/...` dependencies.

### Motivation
Not have to do it ourselves.

### CODE REVIEW

**Review by louis-cqrl — APPROVED** [review]
LGTM

**Review by hush-hush — APPROVED** [review]

File: .github/workflows/update_dependencies.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+          # Generate the PR description
+          cat > $TMP_PR_BODY_PATH <<- EOM
+          ### What does this PR do?
+          Update all `golang.org/x/...` dependencies.
+
+          ### Motivation
+          Keep dependencies up-to-date, benefit from bug fixes and new features.
+
+          ### Describe how you validated your changes
+          CI is considered enough to validate changes.
```
Comment (CelianR) [inline .github/workflows/update_dependencies.yml:72]: **❓ question:** ‏Does it create a PR with empty spaces at the start of each line?

**Review by CelianR — APPROVED** [review]

File: .github/workflows/update_dependencies.yml (hunk 1 above)
Comment (pgimalac) [reply .github/workflows/update_dependencies.yml:71]: I don't think so but I'm about to test it so I'll make sure

Comment (pgimalac) [reply .github/workflows/update_dependencies.yml]: Yep I made sure it was correct, and fixed a couple other issues 😄

**Pull request merged**